serde/src/private/ser.rs

```diff
@@ -58,10 +63,10 @@ enum Unsupported {
     ByteArray,
     Optional,
     Unit,
+    UnitStruct,
```

Comment:
I get a dead code warning in no-default-features mode. Can we cfg away this variant, or is this a bug?

```
$ cargo build --no-default-features

warning: variant is never constructed: `UnitStruct`
  --> serde/src/private/ser.rs:66:5
   |
66 |     UnitStruct,
   |     ^^^^^^^^^^
   |
   = note: #[warn(dead_code)] on by default
```